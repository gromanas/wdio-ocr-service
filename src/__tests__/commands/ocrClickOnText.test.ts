import ocrClickOnText from '../../commands/ocrClickOnText'
import * as OcrGetElementPositionByText from '../../commands/ocrGetElementPositionByText'

jest.mock('../../commands/ocrGetElementPositionByText', ()=>jest.fn())

describe('ocrClickOnText', () => {
  it('should be able to click on text with the correct arguments', async () => {
    const ocrGetElementPositionByTextSpy = jest.spyOn(OcrGetElementPositionByText, 'default')
      .mockResolvedValue({
        searchValue: '',
        matchedString: '',
        originalPosition: { top: 1, bottom: 2, left: 3, right: 4 },
        dprPosition: { top: 1, bottom: 2, left: 3, right: 4 },
        score: 1,
      })
    const globalAny:any = global
    const options = {
      isTesseractAvailable: true,
      ocrImagesPath: 'ocrImagesPath',
      tesseractLang: 'deu',
      reuseOcr: true,
      screenSize: { width: 1, height: 2 },
      text: 'text',
    }

    globalAny.driver = {
      performActions: jest.fn(),
    }

    await ocrClickOnText(options)

    expect(ocrGetElementPositionByTextSpy).toHaveBeenCalledTimes(1)
    expect(globalAny.driver.performActions).toHaveBeenCalledWithSnapshot()
  })

  it('should be able to click on text with a different clickDuration', async () => {
    const globalAny:any = global
    const options = {
      isTesseractAvailable: true,
      ocrImagesPath: 'ocrImagesPath',
      tesseractLang: 'nld',
      reuseOcr: true,
      screenSize: { width: 1, height: 2 },
      text: 'text',
      clickDuration: 250
    }

    globalAny.driver = {
      performActions: jest.fn(),
    }

    await ocrClickOnText(options)

    expect(globalAny.driver.performActions).toHaveBeenCalledWithSnapshot()
  })

  it('should be able to click on text without providing a language', async () => {
    const globalAny:any = global
    const options = {
      isTesseractAvailable: true,
      ocrImagesPath: 'ocrImagesPath',
      reuseOcr: true,
      screenSize: { width: 1, height: 2 },
      text: 'text',
    }

    globalAny.driver = {
      performActions: jest.fn(),
    }

    await ocrClickOnText(options)

    expect(globalAny.driver.performActions).toHaveBeenCalledWithSnapshot()
  })

  it('should be able to click on text with providing a language', async () => {
    const globalAny:any = global
    const options = {
      isTesseractAvailable: true,
      ocrImagesPath: 'ocrImagesPath',
      tesseractLang: 'nld',
      reuseOcr: true,
      screenSize: { width: 1, height: 2 },
      text: 'text',
    }

    globalAny.driver = {
      performActions: jest.fn(),
    }

    await ocrClickOnText(options)

    expect(globalAny.driver.performActions).toHaveBeenCalledWithSnapshot()
  })
})
