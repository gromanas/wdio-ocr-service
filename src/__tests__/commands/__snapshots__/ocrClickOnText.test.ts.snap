// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ocrClickOnText should be able to click on text with a different clickDuration 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Array [
        Object {
          "actions": Array [
            Object {
              "duration": 0,
              "type": "pointerMove",
              "x": 4,
              "y": 2,
            },
            Object {
              "button": 0,
              "type": "pointerDown",
            },
            Object {
              "duration": 250,
              "type": "pause",
            },
            Object {
              "button": 0,
              "type": "pointerUp",
            },
          ],
          "id": "finger1",
          "parameters": Object {
            "pointerType": "touch",
          },
          "type": "pointer",
        },
      ],
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`ocrClickOnText should be able to click on text with the correct arguments 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Array [
        Object {
          "actions": Array [
            Object {
              "duration": 0,
              "type": "pointerMove",
              "x": 4,
              "y": 2,
            },
            Object {
              "button": 0,
              "type": "pointerDown",
            },
            Object {
              "duration": 500,
              "type": "pause",
            },
            Object {
              "button": 0,
              "type": "pointerUp",
            },
          ],
          "id": "finger1",
          "parameters": Object {
            "pointerType": "touch",
          },
          "type": "pointer",
        },
      ],
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
